mailpoet/lib/Newsletter/AutomatedLatestContent.php

```diff
@@ -65,8 +65,13 @@ public function getPosts($args, $postsToExclude = [], $newsletterId = false, $ne
       'post_type' => (isset($args['contentType'])) ? $args['contentType'] : 'post',
       'post_status' => (isset($args['postStatus'])) ? $args['postStatus'] : 'publish',
       'orderby' => 'date',
-      'order' => ($args['sortBy'] === 'newest') ? 'DESC' : 'ASC',
+      'order' => isset($args['sortBy']) && in_array($args['sortBy'], ['newest', 'DESC']) ? 'DESC' : 'ASC',
     ];
+
+    if (!$this->wp->isPostTypeViewable($parameters['post_type'])) {
+      throw new \InvalidArgumentException($this->wp->__('Invalid request parameters', 'mailpoet'));
```

Comment:
I think we should throw here one of the [exceptions in the MailPoet namespace](https://github.com/mailpoet/mailpoet/blob/master/mailpoet%2Flib%2Fexceptions.php). Picking the correct one will also result in the correct HTTP code on the API (which I think should be 4xx, in this case). Now, this would probably result in error 500, which is not desired.

I'd suggest using this one:
```php
throw AccessDeniedException::create()
  ->withErrors([APIError::BAD_REQUEST => __('...', 'mailpoet')]);
```

Also the message could reflect that it's about access denied, I guess.